src/OpenDiffix.Core/QueryEngine.fs

```diff
@@ -15,6 +15,32 @@ let rec private extractColumns query =
 
 type QueryResult = { Columns: Column list; Rows: Row list }
 
+// The main entrypoint to the `OpenDiffix.Core` library.
+// Takes in a SQL `statement` as main parameter along with a set of
+// parameters in `queryContext`.
+//
+// Returns the query result as a list of data rows, along with the
+// columns specifications.
```

Comment:
This is quite obvious a few lines above with `type QueryResult = { Columns: Column list; Rows: Row list }`

Reply:
hm, this is in the vein that "it does that, and doesn't do anything else or return any intermediate result, which just happens to be typed `Row list`, plus `Row` is plain-old-data", which isn't that obvious to anyone doing a first read. I can try to rephrase to emphasis that.